serverless/email/README.md

```diff
@@ -0,0 +1,88 @@
+# email
+
+This is a serverless email webhook, which handles unsubscribes, bounces, idempotency and emailing.
+
+```bash
+.
+├── Makefile                    <-- Make to automate build
+├── README.md                   <-- This instructions file
+├── webhook                     <-- Source code for the webhook handler
+│   ├── main.go                 <-- Lambda function code
+├── unsubscribe                 <-- Source code for the unsubscribe handler
+│   ├── main.go                 <-- Lambda function code
+├── status                      <-- Source code for the sns status reporting handler
+│   ├── main.go                 <-- Lambda function code
+└── template.yaml
+```
+
+## Requirements
+
+* AWS CLI already configured with Administrator permission
+* [Docker installed](https://www.docker.com/community-edition)
+* [Golang](https://golang.org)
+* SAM CLI - [Install the SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html)
+
+## Setup process
+
+### Installing dependencies & building the target 
+
+In this example we use the built-in `sam build` to automatically download all the dependencies and package our build target.   
+Read more about [SAM Build here](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-build.html) 
+
+The `sam build` command is wrapped inside of the `Makefile`. To execute this simply run
+ 
+```shell
+make
+```
+
+### Local development
+
+**Invoking function locally through local API Gateway**
+
+```bash
+sam local start-api
+```
+
+If the previous command ran successfully you should now be able to hit the following local endpoint to invoke your function `http://localhost:3000/hello`
+
+**SAM CLI** is used to emulate both Lambda and API Gateway locally and uses our `template.yaml` to understand how to bootstrap this environment (runtime, where the source code is, etc.) - The following excerpt is what the CLI will read in order to initialize an API and its routes:
+
+```yaml
+...
+Events:
+    HelloWorld:
+        Type: Api # More info about API Event Source: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#api
+        Properties:
+            Path: /hello
+            Method: get
+```
+
+## Packaging and deployment
+
+AWS Lambda Golang runtime requires a flat folder with the executable generated on build step. SAM will use `CodeUri` property to know where to look up for the application:
+
+```yaml
+...
+    FirstFunction:
+        Type: AWS::Serverless::Function
+        Properties:
+            CodeUri: hello_world/
+            ...
+```
+
```

Comment:
Given that Github Actions will deploy this on AWS, we probably want to separate out local testing/development versus Github Actions deployment to sandbox and production.